consensus/consensus-types/src/randomness.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{block::Block, common::Round};
+use aptos_crypto::HashValue;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+struct RandMetadataToSign {
+    epoch: u64,
+    round: Round,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct RandMetadata {
+    metadata_to_sign: RandMetadataToSign,
+    // not used for signing
+    block_id: HashValue,
+    timestamp: u64,
+}
+
+impl From<&Block> for RandMetadata {
+    fn from(block: &Block) -> Self {
+        Self::new(
+            block.epoch(),
+            block.round(),
+            block.id(),
+            block.timestamp_usecs(),
+        )
+    }
+}
+
+impl RandMetadata {
+    pub fn new(epoch: u64, round: Round, block_id: HashValue, timestamp: u64) -> Self {
+        Self {
+            metadata_to_sign: RandMetadataToSign { epoch, round },
+            block_id,
+            timestamp,
+        }
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
```

Comment:
Is this the right way to do? or you didn't notice this

Reply:
copy pasted